openSUSE Tumbleweed Tests

NishkaDG opened the pull request.

sgallagher commented on .travis/opensuse/travis-tasks.sh:
```diff
@@ -0,0 +1,115 @@
+
+set -x
+
+JOB_NAME=${TRAVIS_JOB_NAME:Opensuse tumbleweed}
```
This will take the JOB_NAME directly from the job title in .travis.yml. You will want to make sure that the parser you're using matches. Right now, you have 'tumbleweed' as the fallback default release but you use 'Tumbleweed' as the job name in .travis.yml.  Similarly, you use 'Opensuse' here and 'openSUSE' in the .travis.yml.

I'd prefer if you used the convention of the release portion being lowercase (so 'tumbleweed'). Since the openSUSE project prefers that capitalization, let's use that for the os_name.

sgallagher commented on .travis/opensuse/travis-tasks.sh:
```diff
@@ -0,0 +1,115 @@
+set +e
+ninja -C travis test
+if [ $? != 0 ]; then
+    cat travis_v1/meson-logs/testlog.txt
```
There was a bug in the centos test here that I fixed recently. Make the same fix here:
```suggestion
    cat travis/meson-logs/testlog.txt
```

sgallagher commented on .travis/travis-opensuse.sh:
```diff
@@ -0,0 +1,35 @@
+set -e
+set -x
+
+JOB_NAME=${TRAVIS_JOB_NAME:- Opensuse tumbleweed}
```
Same as above, make sure the spelling matches.

sgallagher commented:
Please also rebase this patch atop the current git master by doing:
```
git remote update
git rebase -i origin/master
```
and resolve any conflicts. The above commands assume that you used the upstream repo as the 'origin' for your checkout. If not, use the correct git remote name in its place.

Conan-Kudo commented on .travis/opensuse/travis-tasks.sh:
```diff
@@ -0,0 +1,115 @@
+# Test the v2 code with clang-analyzer
+# This requires meson 0.49.0 or later
+set +e
+rpmdev-vercmp `meson --version` 0.49.0
```
You don't install `rpmdevtools` for this to work...

Conan-Kudo commented on .travis/opensuse/travis-tasks.sh:
```diff
@@ -0,0 +1,115 @@
+pushd build_rpm
+
+zypper install --no-confirm --no-gpg-checks \
+               --allowerasing \
```
This switch does not exist in Zypper?

Conan-Kudo commented on .travis/opensuse/travis-tasks.sh:
```diff
@@ -0,0 +1,115 @@
+
+zypper install --no-confirm --no-gpg-checks \
+               --allowerasing \
+               -ar -c -n libmodulemd-travis,rpmbuild/RPMS \
```
Are you trying to add a repo here?

sgallagher commented on .travis/opensuse/Dockerfile.deps.tmpl:
```diff
@@ -0,0 +1,31 @@
+	python3-GitPython \
+	python3-gobject \
+	python3-pycodestyle \
+	python3-rpm-macros \
```
This package does not exist in openSUSE and appears to break the test when I try to run it locally with `./.travis/travis-opensuse.sh`

@Conan-Kudo Is there an equivalent package that should replace it?

Conan-Kudo replied:
It's just `python-rpm-macros` in openSUSE.

NishkaDG commented:
Could this be reviewed now?

sgallagher commented:
@NishkaDG Looks like some of the tests pass, but not the installed-RPM ones. It appears you're missing whatever the equivalent of the `rpm-build` package is on openSUSE.

NishkaDG commented:
It seems to be working with the CentOS tests. Could this please be reviewed now?

NishkaDG commented:
> 
> 
> This looks good to me, but can you please squash your changes down into a single commit rebased on top of master?
> 
> Some documentation on how to do this: https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History

Thank you. I think I've done that now, but I'm not sure if this is what you wanted?

sgallagher commented:
@NishkaDG No, you need to rebase and squash all of the commits into a single commit. Right now, this PR contains 25(!) commits for a single logical change.

Please read the document that @Conan-Kudo linked carefully.

NishkaDG commented:
> 
> 
> @NishkaDG No, you need to rebase and squash all of the commits into a single commit. Right now, this PR contains 25(!) commits for a single logical change.
> 
> Please read the document that @Conan-Kudo linked carefully.

That's what the second-last commit is. All the other commits squashed together. I'm not sure how to undo the rest from this PR as well since I pushed the earlier commits already? I couldn't understand how to do that much from the linked document, although I was able to make one more commit that was everything else squashed. Should I make a new PR with just the one commit?

sgallagher commented:
@NishkaDG Use `git rebase -i upstream/master` (replacing `upstream` with whatever the git-remote pointing at the upstream master is). The interactive mode will show you the commands you can use. You need to modify all the commits in that stream and turn them into a single commit. Not keep adding new commits at the end. You need to rewrite the git history.

NishkaDG commented:
What do I force push after that? 

    On Sunday, 24 March, 2019, 7:32:46 pm IST, Stephen Gallagher <notifications@github.com> wrote:  
 
 
@NishkaDG Use git rebase -i upstream/master (replacing upstream with whatever the git-remote pointing at the upstream master is). The interactive mode will show you the commands you can use. You need to modify all the commits in that stream and turn them into a single commit. Not keep adding new commits at the end. You need to rewrite the git history.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.

sgallagher commented:
You need `git push -f origin/master` (or whatever your personal fork is, if it's not `origin`).

NishkaDG commented:
> 
> 
> You need `git push -f origin/master` (or whatever your personal fork is, if it's not `origin`).

I've squashed all the commits and now I can't see the old ones, but force pushing isn't working. It's telling me "Everything up to date".

sgallagher commented:
@NishkaDG You need to provide more information. Can you confirm that `git diff upstream/master` shows your expected changes?

NishkaDG commented:
> 
> 
> @NishkaDG You need to provide more information. Can you confirm that `git diff upstream/master` shows your expected changes?

Yes, it is.

NishkaDG commented:
I had to rebase it upon local as well and then force push just the head to origin/master. It looks like that worked?

Conan-Kudo commented:
@NishkaDG It looks like it worked, though could you amend the commit to remove the " and squash previous commits." from the message? It doesn't make sense now that the commits are squashed in. 😉

NishkaDG commented:
@Conan-Kudo git commit --amend brought back all of the previous commits somehow, so I just rebased and squashed again. Is the current commit message okay?

Conan-Kudo commented:
@NishkaDG It has a couple of spelling errors in the message, but otherwise fine.

`tumbleweed` -> `Tumbleweed` (it's part of the distro title).

`CentOs` -> `CentOS` (correct capitalization for the name).

Fix those and we're good to go! 👍

NishkaDG commented:
> 
> 
> @NishkaDG It has a couple of spelling errors in the message, but otherwise fine.
> 
> `tumbleweed` -> `Tumbleweed` (it's part of the distro title).
> 
> `CentOs` -> `CentOS` (correct capitalization for the name).
> 
> Fix those and we're good to go! 👍

Done.
